src/trace/mod.rs

```diff
@@ -104,18 +132,18 @@ pub trait Trace: BatchReader {
 /// useful for views derived from other sources in ways that prevent the
 /// construction of batches from the type of data in the view (for example,
 /// filtered views, or views with extended time coordinates).
-pub trait BatchReader
+pub trait BatchReader: NumEntries + SizeOf + 'static
 where
     Self: Sized,
 {
     /// Key by which updates are indexed.
-    type Key;
+    type Key: DBData;
```

Comment:
I guess this is where most of the simplifications come from

Reply:
yes